### PR TITLE
Highway Convolution

### PR DESCRIPTION
**Pull request by wballard**
1D and 2D highway network convolutions, added in a sample.

### CODE REVIEW

File: examples/images/convnet_highway_mnist.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Building convolutional network
+network = input_data(shape=[None, 28, 28, 1], name='input')
+#highway convolutions with pooling and dropout
+for i in range(3):
+    for j in [3, 2, 1]: 
+        network = highway_conv_2d(network, 16, j, activation='elu')
+    network = max_pool_2d(network, 2)
+    network = batch_normalization(network)
+    print(network)
```
Comment (aymericdamien) [inline examples/images/convnet_highway_mnist.py]: Can you please remove that print function?

File: examples/images/convnet_mnist.py [hunk 2]
```diff
@@ -46,4 +46,4 @@
 model = tflearn.DNN(network, tensorboard_verbose=0)
 model.fit({'input': X}, {'target': Y}, n_epoch=20,
           validation_set=({'input': testX}, {'target': testY}),
-          snapshot_step=100,  show_metric=True, run_id='convnet_mnist')
```
Comment (aymericdamien) [inline examples/images/convnet_mnist.py]: Also, can you restore 'snapshot_step=100' of the original example

**Comment by aymericdamien** [comment]
Nice update! Everything works perfectly, I just let 2 comments about your code (One to remove a print and another one to restore the deleted argument in convnet mnist example), after you correct it, I will merge your pull request.
Thanks for all your help :)

**Comment by aymericdamien** [comment]
Everything looks fine now, Thanks :)

**Comment by stephenjia** [comment]
@wballard @aymericdamien   I feel a little confused about highway convolution layer. In the original paper, it is y = H(x,WH) T(x,WT) + x (1 − T(x,WT)). 
However, in the tflearn code, it is 
"
convolved = tf.nn.conv2d(incoming, W, strides, padding)
H = activation(convolved + b)
T = tf.sigmoid(tf.mul(convolved, W_T) + b_T)
C = tf.sub(1.0, T)
inference = tf.add(tf.mul(H, T), tf.mul(convolved, C))
"
I thought it should be 
inference = tf.add(tf.mul(H, T), tf.mul(incoming, C))

Can you help me on this?
Thanks a lot.

**Comment by wballard** [comment]
Give it a shot and let me know if that works.

**Comment by stephenjia** [comment]
@wballard  Actually, my confusion is whether your implementation is doing the same thing as mentioned in the original paper, that is, 'y = H(x,WH) T(x,WT) + x (1 − T(x,WT))'. I think your implementation is doing 'y = f(H(x,WH)) T(x,WT) + H(x,WH) (1 − T(x,WT))'.
I have seen an implementation using Lasagne here https://github.com/Lasagne/Lasagne/blob/highway_example/examples/Highway%20Networks.ipynb

**Comment by stephenjia** [comment]
@wballard @aymericdamien I still do not get your code. Your implementation is also different from fomorians's which can found here https://github.com/fomorians/highway-cnn/blob/master/main.py
and also lasagne's example code here https://github.com/Lasagne/Lasagne/blob/highway_example/examples/Highway%20Networks.ipynb

**Comment by wballard** [comment]
Pop in a pull request with your interpretation. This approach worked for
me, in that I was getting better test/train -- but that doesn't mean I did
it right :)
